tools/cldr-code/src/main/java/org/unicode/cldr/test/CheckAnnotations.java

```diff
@@ -103,13 +107,24 @@ public CheckCLDR handleCheck(
                 }
             }
             if (ttsMissing) {
+                final CLDRConfig.Environment env = CLDRConfig.getInstance().getEnvironment();
+                final boolean onSurveyTool =
+                        (env == CLDRConfig.Environment.PRODUCTION
+                                || env == CLDRConfig.Environment.SMOKETEST);
+                final Type mainType =
+                        onSurveyTool
+                                ? CheckStatus.errorType // this is an error in ST
+                                : CheckStatus
+                                        .warningType; // but a warning elsewhere (e.g. UNITTEST) so
+                // we can commit
                 result.add(
                         new CheckStatus()
                                 .setCause(this)
-                                .setMainType(CheckStatus.errorType)
+                                .setMainType(mainType)
                                 .setSubtype(Subtype.ttsAnnotationMissing)
                                 .setMessage(
-                                        "Have keywords but missing the corresponding name (tts) entry"));
+                                        "Have keywords but missing the corresponding name (tts) entry; error vs warning depends on environment {0}",
```

Comment:
"err vs warning depends on environment" won't make sense to vetters and probably isn't needed.